src/test/unit/sanitize.test.js

```diff
@@ -204,4 +206,76 @@ describe('sanitizePII', function() {
       expect(extensions.baseURL.length).toEqual(0);
     }
   });
+
+  it('should not sanitize all the preference values inside preference read markers', function() {
+    const profile = getProfileWithMarkers([
+      [
+        'PreferenceRead',
+        1,
+        {
+          type: 'PreferenceRead',
+          startTime: 0,
+          endTime: 1,
+          prefAccessTime: 0,
+          prefName: 'preferenceName',
+          prefKind: 'preferenceKind',
+          prefType: 'preferenceType',
+          prefValue: 'preferenceValue',
+        },
+      ],
+    ]);
+    const PIIToRemove = getRemoveProfileInformation({
+      shouldRemovePreferenceValues: false,
+    });
+
+    const sanitizedProfile = sanitizePII(profile, PIIToRemove).profile;
+    for (const thread of sanitizedProfile.threads) {
```

Comment:
I still think you can be more assertive in these 2 tests: you _know_ that there's only 1 thread, and you _know_ that there's only 1 marker. So we don't need these loops and if conditions that just obscure the tests.

Reply:
Excellent suggestion. I think that I made the appropriate changes in this push!